.github/workflows/ci.yml

```diff
@@ -110,14 +119,26 @@ jobs:
         key: ${{ runner.os }}-${{ matrix.ghc }}-cabal
 
     - name: Lecture 1 - Doctest
-      continue-on-error: true
+      id: doctest
+      continue-on-error: ${{ env.is_main_repo == 'true' }}
```

Comment:
Thanks for the contribution!

Indeed, it was annoying to see failing PRs in the original repository. The CI is not very flexible, and I believe GitHub actions weren't designed to support this course's specific workflow.

Will it work as intended though?

Currently, the `continue-on-error` is always `true`. This means, that if doctest tests fail, the following unit test suite will run nevertheless, and any person will see errors from both CI runs, not only doctest errors.

With the new approach, it looks like any person will see only doctest errors, right?

Reply:
Whether to continue on doctests (or other tests if there are subsequent ones) was a question I waffled about, but ultimately I think they should not continue on errors for the general case.

When `continue-on-error` is true, subsequent steps will still run, but someone may not _see_ the errors, because it will show up as a passing step in the logs and won't be shown the failing output unless they explicitly know to expand the output from the doctest step.  E.g., see https://github.com/haskell-beginners-2022/exercises/actions/runs/4838794549/jobs/8623457335

An alternative I also thought about was to have the steps always `continue-on-error`, but have another step for the general case to fail if any of the tests failed.  However, while someone should notice that there is a failure, the actual steps containing the test failures would remain collapsed.  This could be dealt with by explaining that they should expand the log entries for specific steps in the failure message, but it seems less friendly to the user than having the failure output expanded.

(On a side note, it could be more friendly to the learner to separate out the build step from the test step, so that the learner only needs to look at the test output without having to skip through the build output, unless there actually is a build error.  I'd be happy to contribute a separate pull request if you think this is a good idea.)